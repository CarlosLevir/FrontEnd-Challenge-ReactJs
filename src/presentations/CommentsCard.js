import React from 'react';
import { Panel } from 'react-bootstrap';

const styles = {
  item: {
    height: 200,
    maxWidth: 400,
    margin: 5
  },
  bodyPanel: {
    display: 'flex',
    flexWrap: 'wrap'
  }
};

const Comments = ({ comment }) => (
  <div>
    <Panel bsSize="small" style={styles.item}>
      <Panel.Heading>{comment.name}</Panel.Heading>
      <Panel.Body style={styles.bodyPanel}>
        Email:
        {' '}
        {comment.email}
        {' '}
        {comment.body}
      </Panel.Body>
    </Panel>
  </div>
);

export default Comments;
