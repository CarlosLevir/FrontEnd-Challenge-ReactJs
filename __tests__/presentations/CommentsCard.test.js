import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import CommentsCard from '../../src/presentations/CommentsCard';

const comment = {
  name: 'test',
  email: 'test@test',
  title: 'test',
  body: 'test'
};

configure({ adapter: new Adapter() });

describe('CommentsCard', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<CommentsCard comment={comment} />).shallow();
    expect(wrapper).toMatchSnapshot();
  });
});
