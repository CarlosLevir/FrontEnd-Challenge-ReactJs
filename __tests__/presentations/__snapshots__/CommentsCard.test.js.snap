// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsCard renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <div>
    <Uncontrolled(Panel)
      bsSize="small"
      style={
        Object {
          "height": 200,
          "margin": 5,
          "maxWidth": 400,
        }
      }
    >
      <PanelHeading
        bsClass="panel"
        componentClass="div"
      >
        test
      </PanelHeading>
      <PanelBody
        bsClass="panel"
        collapsible={false}
        style={
          Object {
            "display": "flex",
            "flexWrap": "wrap",
          }
        }
      >
        Email:
         
        test@test
         
        test
      </PanelBody>
    </Uncontrolled(Panel)>
  </div>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Uncontrolled(Panel)
        bsSize="small"
        style={
          Object {
            "height": 200,
            "margin": 5,
            "maxWidth": 400,
          }
        }
      >
        <PanelHeading
          bsClass="panel"
          componentClass="div"
        >
          test
        </PanelHeading>
        <PanelBody
          bsClass="panel"
          collapsible={false}
          style={
            Object {
              "display": "flex",
              "flexWrap": "wrap",
            }
          }
        >
          Email:
           
          test@test
           
          test
        </PanelBody>
      </Uncontrolled(Panel)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsSize": "small",
        "children": Array [
          <PanelHeading
            bsClass="panel"
            componentClass="div"
          >
            test
          </PanelHeading>,
          <PanelBody
            bsClass="panel"
            collapsible={false}
            style={
              Object {
                "display": "flex",
                "flexWrap": "wrap",
              }
            }
          >
            Email:
             
            test@test
             
            test
          </PanelBody>,
        ],
        "style": Object {
          "height": 200,
          "margin": 5,
          "maxWidth": 400,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "panel",
            "children": "test",
            "componentClass": "div",
          },
          "ref": null,
          "rendered": "test",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "panel",
            "children": Array [
              "Email:",
              " ",
              "test@test",
              " ",
              "test",
            ],
            "collapsible": false,
            "style": Object {
              "display": "flex",
              "flexWrap": "wrap",
            },
          },
          "ref": null,
          "rendered": Array [
            "Email:",
            " ",
            "test@test",
            " ",
            "test",
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Uncontrolled(Panel)
          bsSize="small"
          style={
            Object {
              "height": 200,
              "margin": 5,
              "maxWidth": 400,
            }
          }
        >
          <PanelHeading
            bsClass="panel"
            componentClass="div"
          >
            test
          </PanelHeading>
          <PanelBody
            bsClass="panel"
            collapsible={false}
            style={
              Object {
                "display": "flex",
                "flexWrap": "wrap",
              }
            }
          >
            Email:
             
            test@test
             
            test
          </PanelBody>
        </Uncontrolled(Panel)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsSize": "small",
          "children": Array [
            <PanelHeading
              bsClass="panel"
              componentClass="div"
            >
              test
            </PanelHeading>,
            <PanelBody
              bsClass="panel"
              collapsible={false}
              style={
                Object {
                  "display": "flex",
                  "flexWrap": "wrap",
                }
              }
            >
              Email:
               
              test@test
               
              test
            </PanelBody>,
          ],
          "style": Object {
            "height": 200,
            "margin": 5,
            "maxWidth": 400,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "panel",
              "children": "test",
              "componentClass": "div",
            },
            "ref": null,
            "rendered": "test",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "panel",
              "children": Array [
                "Email:",
                " ",
                "test@test",
                " ",
                "test",
              ],
              "collapsible": false,
              "style": Object {
                "display": "flex",
                "flexWrap": "wrap",
              },
            },
            "ref": null,
            "rendered": Array [
              "Email:",
              " ",
              "test@test",
              " ",
              "test",
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
